lite/src/sequential.rs

```diff
@@ -0,0 +1,102 @@
+use crate::types::*;
+
+/// SequentialVerifier verifies all headers in order,
+/// where +2/3 of the correct validator set must have signed the header.
+struct SequentialVerifier<H, V>
+where
+    H: Header,
+    V: Validators,
+{
+    trusting_period: Time,
+    state: TrustedState<H, V>,
+}
+
+impl<H, V> SequentialVerifier<H, V>
+where
+    H: Header,
+    V: Validators,
+{
+    /// Returns the time after which the current state expires
+    /// and the verifier must be reset subjectively.
+    fn expires(&self) -> Time {
+        self.state.header.bft_time() + self.trusting_period
+    }
+
+    /// Verify takes a header, a commit for the header,
+    /// and the next validator set referenced by the header.
+    /// Without knowing this next validator set, we can't
+    /// really verify the next header, so we make verifying
+    /// the current header conditional on receiving that validator set.
+    /// We could break this up into two steps, one where we verify the header,
+    /// and one where we receive and validate the next validators,
+    /// but for now we combine them. Returns an error if verification fails.
+    fn verify<C>(&self, now: Time, header: H, commit: C, next_validators: V) -> Result<(), Error>
```

Comment:
Why we need next_validators here when we have it as `TrustedState.next_validators`?

Reply:
`next` is confusing :(

The `trusted_state.next_validators` is with respect to `trusted_state.header`.

So if the `trusted_state.header` is H, the `trusted_state.next_validators` is for H+1. but the `next_validators` we receive in this verify method are actually for `H+2` ...

Reply:
Can we find a better name for `next_validators` s.t. it's less confusing then? Similarly with `header`. 
Maybe `next_header` and `next_validators_for_header` (if that's right). That would indicate that: we are expecting the next header in the sequence.